hbp_validation_framework/__init__.py

```diff
@@ -308,53 +996,734 @@ def get_platform(self):
+        in the following ways (in order of priority):
 
-    # List models
-    models = model_library.list_models(brain_region="hippocampus")
+        1. specify the `model_id`
+        2. specify the `alias` (of the model)
 
-    # Download a model description
-    model_description = model_library.get_model(model_uri)
+        Parameters
+        ----------
+        model_id : UUID
+            System generated unique identifier associated with model description.
+        alias : string
+            User-assigned unique identifier associated with model description.
+        instances : boolean, optional
+            Set to False if you wish to omit the details of the model instances; default True.
+        images : boolean, optional
+            Set to False if you wish to omit the details of the model images; default True.
 
-    """
+        Returns
+        -------
+        dict
+            Entire model description as a JSON object.
+
+        Examples
+        --------
+        >>> model = model_catalog.get_model(model_id="8c7cb9f6-e380-452c-9e98-e77254b088c5")
+        >>> model = model_catalog.get_model(alias="B1")
+        """
+
+        if model_id == "" and alias == "":
+            raise Exception("Model ID or alias needs to be provided for finding a model.")
+        elif model_id != "":
+            url = self.url + "/scientificmodel/?id=" + model_id + "&format=json"
+        else:
+            url = self.url + "/scientificmodel/?alias=" + alias + "&format=json"
 
-    def get_model(self, model_uri):
-        model = requests.get(self.url + model_uri, auth=self.auth).json()
-        return model
+        model = requests.get(url, auth=self.auth)
+        if str(model) != "<Response [200]>":
+            raise Exception("Error in retrieving model. Response = " + str(model))
+        model = model.json()
+        if len(model["models"]) == 0:
+            raise Exception("Error in retrieving model. Possibly invalid model_id or alias. Response = " + str(model))
+        if instances == False:
+            model["models"][0].pop("instances")
+        if images == False:
+            model["models"][0].pop("images")
+        return model["models"][0]
 
     def list_models(self, **filters):
-        model_list_uri = self.url + "/models/"  # todo: support filters
-        models = requests.get(model_list_uri, auth=self.auth).json()
-        return models
-
-    def register(self, name, description="",  species="", brain_region="",
-                 cell_type="", author="", source=""):
-        data = locals()
-        data.pop("self")
-        model_list_uri = self.url + "/models/"
-        response = requests.post(model_list_uri, data=json.dumps(data),
-                                 auth=self.auth)
-        return response.json()
+        """Retrieve list of model descriptions satisfying specified filters.
+
+        The filters may specify one or more parameters that belong
```

Comment:
"attributes" rather than "parameters"

and include a list of possible attribute names

Reply:
Done.